### PR TITLE
Unit tests for validating absent probes via directory symlinks

### PR DESCRIPTION
**Pull request by almiliMSFT**
Some unit tests that validate that our current implementation does the right thing w.r.t. tracking absent file probes via directory symlinks.  For example:
  for layout: 
```
{rootDir}
└── Versions 
    ├── A
    │   └── file
    └── sym-A -> A
```
and probe `Versions/sym-A/absent-file`, paths that **must** be recorded are
  - `Versions/sym-A`
  - `Versions/A/absent-file`

and path that **must not** be recorded is
  - `Versions/sym-A/absent-file`

### CODE REVIEW

File: Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/DirectorySymlinkTests.cs [hunk 1]
```diff
@@ -374,9 +400,11 @@ public void DirectorySymlinksInOutputDirectory(SealDirectoryKind dirKind)
                     .AssertCacheHit(allPipIds.Except(new[] { consumer.pip.PipId }).ToArray());
             }
 
-            // invalidate producer pip, rerun, expect all cache misses
+            // invalidate producer pip, rerun, expect reads to be cache misses and absent probes to be cache hits 
             InvalidatePip(producerPip);
-            RunSchedulerAndValidateProducedLayout().AssertCacheMiss(allPipIds);
+            RunSchedulerAndValidateProducedLayout()
+                .AssertCacheMiss(readConsumers.Select(t => t.pip.PipId).ToArray())
+                .AssertCacheHit(absentProbeConsumers.Select(t => t.pip.PipId).ToArray());
```
Comment (mpysson) [inline Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/DirectorySymlinkTests.cs]: >absentProbeConsumers [](start = 32, length = 20)

add a test case where these files become existing as discussed. Thanks for double checking test coverage on this!

File: Examples/DotNetCoreBuild/config.dsc [hunk 2]
```diff
@@ -5,7 +5,7 @@ config({
     modules: [
         d`sdk`,
         d`src`,
-        d`test`
+        d`test`,
```
Comment (almiliMSFT) [inline Examples/DotNetCoreBuild/config.dsc]:
```suggestion
        d`test`
```

File: Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/DirectorySymlinkTests.cs (hunk 1 above)
Comment (almiliMSFT) [reply Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/DirectorySymlinkTests.cs]: added (lines 518 and below)

File: Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/DirectorySymlinkTests.cs [hunk 3]
```diff
@@ -471,6 +513,32 @@ public void DirectorySymlinksInSealSourceDirectory(SealDirectoryKind sealKind)
...
+            XAssert.All(AbsentProbeSpecs, spec => !File.Exists(X($"{sealDir}/{spec.Lookup}")));
+
+            // create all absent lookup paths
+            foreach (var spec in AbsentProbeSpecs)
+            {
+                var path = X($"{sealDir}/{spec.Lookup}");
+                File.WriteAllText(path, "text");
+            }
+
+            // assert that all lookup paths that were absent are not present
```
Comment (almiliMSFT) [inline Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/DirectorySymlinkTests.cs]:
```suggestion
            // assert that all lookup paths that were absent are now present
```

**Comment by almiliMSFT** [comment]
Some unit tests that validate that our current implementation does the right thing w.r.t. tracking absent file probes via directory symlinks. For example:
for layout:

```
{rootDir}
└── Versions 
    ├── A
    │   └── file
    └── sym-A -> A
```

and probe Versions/sym-A/absent-file, paths that must be recorded are

    Versions/sym-A
    Versions/A/absent-file

and path that must not be recorded is

    Versions/sym-A/absent-file

**Pull request merged**